### DISCUSSION TITLE
CC2652P ZigBee2mqtt USB Stick doesnt work

### DISCUSSION THREAD

**Discussion by 82timo** [Koenkk/zigbee2mqtt#10811] [Devices]
Hi,

I bought a CC2652P ZigBee2mqtt USB Stick, but I dont get ot managed to work unter openhabian. The console gives some errors and I dont know how to fix them.

1. The Stick is flashed with Koenkk 3.x.0 (CC1352P2_CC2652P_launchpad_coordinator_20211217)
2. I installed mosquitto. It seems to work fine:
`openhabian@openhabian:~ $ sudo service mosquitto status
● mosquitto.service - Mosquitto MQTT Broker
     Loaded: loaded (/lib/systemd/system/mosquitto.service; enabled; vendor preset: enabled)
     Active: active (running) since Sat 2022-01-15 11:36:40 CET; 4min 10s ago
       Docs: man:mosquitto.conf(5)
             man:mosquitto(8)
    Process: 634 ExecStartPre=/bin/mkdir -m 740 -p /var/log/mosquitto (code=exited, status=0/SUCCESS)
    Process: 636 ExecStartPre=/bin/chown mosquitto /var/log/mosquitto (code=exited, status=0/SUCCESS)
    Process: 637 ExecStartPre=/bin/mkdir -m 740 -p /run/mosquitto (code=exited, status=0/SUCCESS)
    Process: 638 ExecStartPre=/bin/chown mosquitto /run/mosquitto (code=exited, status=0/SUCCESS)
   Main PID: 639 (mosquitto)
      Tasks: 1 (limit: 4915)
        CPU: 169ms
     CGroup: /system.slice/mosquitto.service
             └─639 /usr/sbin/mosquitto -c /etc/mosquitto/mosquitto.conf`

3. I inserted the stick in a free USB Slot on my raspberry pi (I tried all 4) and checked the direction:

`openhabian@openhabian:~ $ ls -l /dev/ttyUSB0
crw-rw---- 1 root dialout 188, 0 Jan 15 11:37 /dev/ttyUSB0`

**/dev/ttyUSB0** is right

4. I installed nodejs:
`sudo apt-get install -y nodejs npm git make g++ gcc`

5. I checked the installation:
`openhabian@openhabian:~ $ node --version
v14.18.3
openhabian@openhabian:~ $ npm --version
6.14.15`

Installation went right.

6. I installed zigbeeMQTT
`git clone https://github.com/Koenkk/zigbee2mqtt.git`

`sudo mv zigbee2mqtt /opt/zigbee2mqtt`

`cd /opt/zigbee2mqtt`

`npm ci`

7.
I opened the config file (configuration.yaml):
`nano /opt/zigbee2mqtt/data/configuration.yaml`

8. I changed the content of the config file

`mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://mqtt
  user: openhabian
  password: MYPASSWORD <- my individual mosquitto password
serial:
  port: /dev/ttyUSB0
frontend:
  port: 8080
advanced:
  network_key: GENERATE`

I saved the file with STRG+O and confirmed enter (File Name to Write: /opt(zigbee2mqtt/data/configuration.yaml).
Then I closed GNU nano 5.4 wth STRG+X and went back to the terminal.

9. I gave the rights for the direction opt/zigbee2mqtt to openhabian:
`sudo chown -R openhabian:openhabian /opt/zigbee2mqtt`

10. I changed the direction
`cd /opt/zigbee2mqtt`

11. I startet mqtt
`npm start`

12. The console gives me the following errors:
`openhabian@openhabian:/opt/zigbee2mqtt $ npm start

> zigbee2mqtt@1.22.2 start /opt/zigbee2mqtt
> node index.js

Zigbee2MQTT:info  2022-01-15 12:51:40: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2022-01-15.12-51-38' filename: log.txt
Zigbee2MQTT:info  2022-01-15 12:51:40: Starting Zigbee2MQTT version 1.22.2 (commit #301ba14b)
Zigbee2MQTT:info  2022-01-15 12:51:40: Starting zigbee-herdsman (0.13.188)
Zigbee2MQTT:error 2022-01-15 12:51:40: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-01-15 12:51:40: Failed to start zigbee
Zigbee2MQTT:error 2022-01-15 12:51:40: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-01-15 12:51:40: Exiting...
Zigbee2MQTT:error 2022-01-15 12:51:41: Error: Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'
    at SerialPort.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/znp/znp.ts:146:28)
    at SerialPort._error (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/node_modules/@serialport/stream/lib/index.js:198:14)
    at /opt/zigbee2mqtt/node_modules/zigbee-herdsman/node_modules/@serialport/stream/lib/index.js:242:12
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.22.2 start: `node index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the zigbee2mqtt@1.22.2 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/openhabian/.npm/_logs/2022-01-15T11_51_41_060Z-debug.log`

What have I done wrong/what is missing?

Best regards,
Timo

**Comment by Koenkk** [comment]
> 'Error: Error Resource temporarily unavailable Cannot lock port'

Another process is locking the stick: https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html#error-resource-temporarily-unavailable-cannot-lock-port

**Reply by 82timo** [reply]
Thank you for your reply.

`openhabian@openhabian:~ $ ls -l /proc/[0-9]/fd/ |grep /dev/ttyUSB0
ls: cannot open directory '/proc/1/fd/': Permission denied
ls: cannot open directory '/proc/2/fd/': Permission denied
ls: cannot open directory '/proc/3/fd/': Permission denied
ls: cannot open directory '/proc/4/fd/': Permission denied
ls: cannot open directory '/proc/7/fd/': Permission denied
ls: cannot open directory '/proc/8/fd/': Permission denied
ls: cannot open directory '/proc/9/fd/': Permission denied
`

**Reply by m0wlheld** [reply]
Must be run as root…

**Reply by 82timo** [reply]
How?
openhabian@openhabian:~ $ sudo ls -l /proc/[0-9]/fd/ |grep /dev/ttyUSB0

Gives no answer

**Comment by lux73** [comment]
> Error while opening serialport

make sure your User is added to the Group "dialout"

https://askubuntu.com/questions/112568/how-do-i-allow-a-non-default-user-to-use-serial-device-ttyusb0

**Reply by 82timo** [reply]
openhabian@openhabian:~ $ sudo adduser openhabian  dialout
The user `openhabian' is already a member of `dialout'.


Seems that this is no solution for me.
